include/RAJA/policy/cuda/forall.hpp

```diff
@@ -78,50 +78,78 @@ struct ForallDimensionCalculator<::RAJA::iteration_mapping::Direct,
                                  ::RAJA::cuda::IndexGlobal<dim, BLOCK_SIZE, GRID_SIZE>,
                                  UniqueMarker>
 {
-  static_assert(BLOCK_SIZE > 0 || BLOCK_SIZE == named_usage::unspecified, "block size may not be ignored with forall");
-  static_assert(GRID_SIZE > 0 || GRID_SIZE == named_usage::unspecified, "grid size may not be ignored with forall");
+  static_assert(BLOCK_SIZE > 0, "block size may not be ignored with forall");
```

Comment:
```suggestion
  static_assert(BLOCK_SIZE > 0, "block size > 0 must be specified with forall");
```
Is this more clear? Similarly, with the following static assert.

Reply:
These comments are a bit cryptic now that you point it out. There are currently four possible states, positive, 0, -1 and "other." Positive means a compile time size. 0 means unspecified and is handled in another partial specialization. -1 means ignored and is not allowed in forall as you must use blocks and threads. "Other" doesn't have a meaning and is always invalid. This static assert is mainly guarding against -1/ignored but also guards against "other" values.

Reply:
So positive and 0 values are valid in forall. I was trying to call out that ignored is not a valid value here.